tests/js/common/http/grants-spec.js

```diff
@@ -134,3 +135,216 @@ describe('Grants', function() {
     expect(JSON.parse(resp.body)).to.have.property('result', 'rw');
   });
 });
+
+describe('UserProperties', function () {
+  const rootUser = 'root';
+  const rootDB = '_system';
+
+  const nonRootDB = 'nonRootDB';
+  const nonRootUser = 'notRoot';
+
+  const configPropertyAttribute = 'graphs';
+  const additionalPropertyAttribute = 'someAdditionalAttribute';
+
+  const initConfigObject = {
+    a: true,
+    b: false,
+    c: 1,
+    d: 1.1,
+    e: "aString",
+    f: [1, 2, 3],
+    g: {"anObjectKey": "anObjectValue"}
+  };
+
+  const overwriteConfigObject = {
+    c: 1337,
+    someNewParam: "MJ7"
+  };
+
+  const mergeConfigObject = {
+    "thisMustStay": true
+  };
+
+  const verifyEmptyConfig = (result) => {
+    console.warn(result);
+    expect(result.error).to.be.false;
+    expect(result).to.have.property('code', 200);
+    expect(result).to.have.property('result', null);
+  };
+
+  const verifyInitObject = (result) => {
+    expect(result.error).to.be.false;
+    expect(result).to.have.property('code', 200);
+
+    result = result.result[configPropertyAttribute];
+    expect(result.a).to.be.true;
+    expect(result.b).to.be.false;
+    expect(result).to.have.property('c', 1);
+    expect(result).to.have.property('d', 1.1);
+    expect(result).to.have.property('e', "aString");
+    expect(result.f).to.be.an('array');
+    expect(result.g).to.be.an('object');
+  };
+
+  const verifyOverwriteConfigObject = (result) => {
+    expect(result.error).to.be.false;
+    expect(result).to.have.property('code', 200);
+
+    result = result.result[configPropertyAttribute];
```

Comment:
Should this also verify the "old", non-overwritten attributes (e.g. 'a', 'b', 'c', 'd', 'e', 'f', 'g')?
I understand that overwriting data is just _merging_ values together, so the previous data in non-touched attributes should remain unchanged.

Reply:
No - this might be a bit misleading. Maybe I should add another comment to this. 
First we initialize "graphs" object, then we overwrite the "graphs" object completely. Then we "merge/ add another attribute to the top level next to graphs".

Reply:
Ok. I think adding a code comment could help indeed.

Reply:
Added